src/main/java/duke/utilities/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.utilities;
+
+import duke.commands.*;
+import duke.tasks.*;
+import duke.exceptions.*;
+import java.util.regex.Pattern;
+
+public class Parser {
+    public Parser() {
+
+    }
+
+    public static String parseTask(Task task) { // parses task into string to load into storage
+        String parsed;
+        TaskType taskType = task.getTaskType();
+        switch (taskType) {
+            case TODO:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription();
+                break;
+            case EVENT:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription() + "|" + ((Event) task).getTaskTime();
+                break;
+            case DEADLINE:
+                parsed = taskType.toString() + "|" + task.getDoneInt() + "|" + task.getDescription() + "|" + ((Deadline) task).getTaskTime();
+                break;
+            default:
+                parsed = ""; // TODO add error message, change duke.exceptions.DukeException to such task exists
+        }
+        return parsed;
+    }
+
+    public static Task parseFile(String line) { // parses line from tasks.txt into a task
+        String[] split = line.split(Pattern.quote("|"));
+        switch (split[0]) {
```

Comment:
I also face this problem in intelliJ due to its auto fixed indentations. There should not be indentation for the "case" lines

Reply:
Oh you can highlight the paragraphs and press shift and tab together to undo the indentation!